File: bench/selection-sort.lua
function sortDescending(arr)
    local len = #arr
    for i=1,len do
        for j=i+1,len do
            if (arr[i] < arr[j]) then
                local t = arr[i]
                arr[i] = arr[j]
                arr[j] = t
            end
        end
    end
end

local time = os.clock()
local arr = {}
for i=1,4000 do
    arr[i] = i-1
end
sortDescending(arr)
print(arr[1])
print((os.clock()-time)*1000)

File: bench/selection-sort.js
function sortDescending(arr) {
    let len = arr.length
    for (let i = 0; i < len; i++) {
        for (let j = i + 1; j < len; j++) {
            if (arr[i] < arr[j]) {
                let t = arr[i]
                arr[i] = arr[j]
                arr[j] = t
            }
        }
    }
}

let time = performance.now()
let arr = new Array(4000)
for (let i = 0; i < arr.length; i++) {
    arr[i] = i
}
sortDescending(arr)
console.log(arr[0])
console.log(performance.now() - time)